test.py

```diff
@@ -1360,14 +1360,25 @@ def write_junit_report(tmpdir: str, mode: str) -> None:
 
 
 def write_consolidated_boost_junit_xml(tmpdir: str, mode: str) -> None:
-    xml = ET.Element("TestLog")
+    # collects all suites grouped by their names
+    test_suites_with_name = {}
     for suite in TestSuite.suites.values():
         for test in suite.tests:
             if test.mode != mode:
                 continue
             test_xml = test.get_junit_etree()
-            if test_xml is not None:
-                xml.extend(test_xml.getroot().findall('.//TestSuite'))
+            if test_xml is None:
+                continue
+            test_suites = test_xml.getroot().findall('.//TestSuite')
```

Comment:
How is it possible that we can have more than one test suite node in a single test?

Reply:
Agreed. But it’s not in the scope of this change. We can surely apply your suggestion in another PR.

Reply:
i tested locally, we do have multiple test suites node in a single test: if we enumerate the tests in a given suite whose "name" is "boost", we would have multiple "TestCase" tags, whose parent tag are "boost.Master Test Suite.debug", but sometimes if the test has its "BOOST_TEST_MODULE" defined, the test suite tag's name would be "boost.<the-value-of-BOOST_TEST_MODULE>.debug". in this case, we are testing with the debug build.